### PR TITLE
Handle unpinning failure

### PR DESCRIPTION
**Pull request by mbrizic**

### CODE REVIEW

File: nodes/persistence-node/src/types/TrackedIpfsHashInfo.ts [hunk 1]
```diff
@@ -2,9 +2,17 @@ import { UnresponsiveIpfsHashInfo } from "./UnresponsiveIpfsHashInfo";
...
-  isWrapper?: boolean;
-  isPinned: boolean;
+  status: Status;
+  previousStatus?: Status;
   unresponsiveInfo?: UnresponsiveIpfsHashInfo;
   indexes: string[];
-  isLost?: boolean;
 };
+
+export enum Status {
```
Comment (nerfZael) [inline nodes/persistence-node/src/types/TrackedIpfsHashInfo.ts:11]: Rename to TrackedIpfsHashStatus and move to separate file

File: nodes/persistence-node/src/services/persistence-service/PersistenceService.ts [hunk 2]
```diff
@@ -135,73 +135,74 @@ export class PersistenceService {
...
       });
     } else if (result === "timeout") {
-      await this.scheduleRetry(ipfsHash, retryCount, indexes);
+      await this.scheduleRetry(ipfsHash, retryCount, Status.CheckingIfWrapper, indexes);
     }
   }
   
   private async tryUntrackIpfsHash(info: TrackedIpfsHashInfo): Promise<void> {
-    if(!info.isWrapper) {
+    if(info.status === Status.NotAWrapper) {
```
Comment (nerfZael) [inline nodes/persistence-node/src/services/persistence-service/PersistenceService.ts:155]: If the hash is not pinned we can just remove it. So it'll be:
`if(info.status !== Status.Pinned && info.status !== Status.Unpinning)`

File: nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts [hunk 3]
```diff
@@ -92,7 +92,8 @@ const calculateCIDsToUntrack = (
     //If the IPFS hash is not in any index
     if(!cidIndexesMap[info.ipfsHash]) {
       //Untrack the IPFS hash unless the index for which it was previously logged for is not able to be retrieved
-      if(!info.indexes.some(x => unresponsiveIndexMap[x])) {
+      // and if it's not considered lost
+      if(!info.indexes.some(x => unresponsiveIndexMap[x]) && info.status !== Status.Lost) {
```
Comment (nerfZael) [inline nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts:96]: If the package is lost then we will forever keep a record of it? That doesn't sound right

**Review by nerfZael — CHANGES_REQUESTED** [review]

File: nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts (hunk 3 above)
Comment (mbrizic) [reply nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts:96]: Ok, if we don't need to care about them, then do we even need this part:

https://github.com/polywrap/nodes/pull/128/files#diff-c8ab1fb7d43f3b45c4468d7ecf0ce060d8003c30065893bbcc7734e4b567b2c6L199

Should we just remove them immediately at that point?

Comment (mbrizic) [reply nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts:96]: (and then we maybe don't even need a `Lost` status)

Comment (nerfZael) [reply nodes/persistence-node/src/services/persistence-service/utils/calculateCIDsToTrackAndUntrack.ts:96]: We need to keep track of all hashes that the indexers have. If we untracked a hash when it is lost, then the next time we query the indexers (every 15 secs) we would get that hash back and would have to re-process is again (retry until lost), and again..

**Pull request merged**